parlai/scripts/self_chat.py

```diff
@@ -86,15 +92,22 @@ def _run_self_chat_episode(opt, world, world_logger):
 
 def self_chat(opt):
     random.seed(opt['seed'])
+    partner = opt['partner']
 
     # Create agents
     agent1 = create_agent(opt, requireModelExists=True)
-    agent2 = agent1.clone()
-
-    # Set IDs
     model_id = agent1.id
-    agent1.id = model_id + "_1"
-    agent2.id = model_id + "_2"
+    if partner:
+        # self chat with different models
+        opt['model_file'] = partner
+        agent2 = create_agent(opt, requireModelExists=True)
+        model_id = agent1.id + "_" + agent2.id
+    else:
+        # self chat with same model
+        agent2 = agent1.clone()
+        # Set IDs
+        agent1.id = model_id + "_1"
```

Comment:
shouldn't this be out of the if one indentation back?  or else shouldn't agent1.id  and agent2.id be defined for the other case?
i think you still need "_1" and "_2" even for the partner case in case they are the same model?

Reply:
Ah, but the only case where we want to set `agentX.id` is when they're twins right? because we modify it to add `_1` and `_2`, otherwise they don't *have* to be defined, id is set from the model being loaded

Reply:
As I said in comment,  I would still put "_1" and "2_" on them, because they might have the same name

Reply:
I see your point, fixed!